buildSrc/src/main/groovy/org/elasticsearch/gradle/test/DistroTestPlugin.java

```diff
@@ -164,9 +177,30 @@ private void configureCopyPackagingTask(Project project) {
         this.copyPackagingArchives = Boilerplate.maybeRegister(project.getParent().getTasks(), COPY_PACKAGING_TASK, Copy.class,
             t -> {
                 t.into(archivesDir);
-                t.from(project.getConfigurations().getByName(PACKAGING_DISTRIBUTION));
+                t.from(project.getConfigurations().getByName(PACKAGING_CONFIGURATION));
 
                 Path archivesPath = archivesDir.get().getAsFile().toPath();
+                TaskInputs inputs = t.getInputs();
+                inputs.property("version", VersionProperties.getElasticsearch());
+                t.doLast(action -> {
+                    try {
+                        Files.writeString(archivesPath.resolve("version"), VersionProperties.getElasticsearch());
+                    } catch (IOException e) {
+                        throw new UncheckedIOException(e);
+                    }
+                });
+        });
+    }
+
+    private void configureCopyUpgradeTask(Project project) {
+        this.upgradeDir = project.getParent().getLayout().getBuildDirectory().dir("packaging/upgrade");
+        // temporary, until we have tasks per distribution
```

Comment:
Does this mean we intend to ditch this assumption that all distro test projects will be under a common parent? Why not just use a similar pattern as we do elsewhere and add this to the root project so there's no risk of there ever being more than one?

Reply:
This is already in the root project. All of these copy tasks, and their associated target directories, are part of qa/vagrant.